apps/checker/app/services/EntityHelper.scala

```diff
@@ -0,0 +1,63 @@
+package services
+
+import com.gu.typerighter.model.TextRange
+import opennlp.tools.namefind.{NameFinderME, TokenNameFinderModel}
+import opennlp.tools.tokenize.SimpleTokenizer
+
+import java.io.InputStream
+
+case class EntityInText(word: String, range: TextRange, model: String)
+
+class EntityHelper() {
+  val personInputFile: InputStream =
+    getClass.getClassLoader.getResourceAsStream("resources/opennlp/en-ner-person.bin")
+  val personModel = new TokenNameFinderModel(personInputFile)
+  val personFinder = new NameFinderME(personModel)
+
+  val organizationInputFile: InputStream =
+    getClass.getClassLoader.getResourceAsStream("resources/opennlp/en-ner-organization.bin")
+  val organizationModel = new TokenNameFinderModel(organizationInputFile)
+  val organizationFinder = new NameFinderME(organizationModel)
+
+  val locationInputFile: InputStream =
+    getClass.getClassLoader.getResourceAsStream("resources/opennlp/en-ner-location.bin")
+  val locationModel = new TokenNameFinderModel(locationInputFile)
+  val locationFinder = new NameFinderME(locationModel)
+
+  def getAllEntitiesFromText(text: String): List[EntityInText] = {
+    getEntitiesFromText(text, "person") ++
+      getEntitiesFromText(text, "organization") ++
+      getEntitiesFromText(text, "location")
+  }
+
+  def getEntitiesFromText(text: String, model: String): List[EntityInText] = {
+    val finder = model match {
```

Comment:
Non-blocking: rather than pattern match on a `String` type here, which will never be exhaustive (hence the match that throws), I'd be tempted to put all the finders in a list, and iterate across that – something like

```scala
// In the class constructor
val allFinders = List(personFinder, ...)

// In `getEntitiesFromText`
finders.foldLeft(List.empty[EntityInText])((finder, entities) => {
  val entitySpans = finder.find(tokenSpans.map(_.getCoveredText(text).toString))
  // ... accumulate entities
})
```

feel like that'll remove an opportunity for an error to be thrown and might be fewer lines – WDYT?

Reply:
Good shout, this didn't feel great before, I've made that change here: https://github.com/guardian/typerighter/pull/443/commits/e0fa022c13e8829eb17b174a3510accd6298a380

It does mean we lose the `model` attribute but I think that's fine - we weren't using it for anything anyway.